runtime/calamari/src/diff_gas_fees.rs

```diff
@@ -0,0 +1,2169 @@
+    module: String,
+    extrinsic: String,
+    gas_fee: f64,
+}
+
+fn get_call_details(call: &crate::Call) -> (DispatchInfo, u32) {
+    let dispatch_info =
+        <<Runtime as frame_system::Config>::Call as GetDispatchInfo>::get_dispatch_info(call);
+    let call_len = call.using_encoded(|e| e.len()) as u32;
+    (dispatch_info, call_len)
+}
+
+#[test]
+fn diff_gas_fees() {
+    const CURRENT_PATH: &str = env!("CARGO_MANIFEST_DIR");
+    let mut latest_version = String::new();
+    for file in std::fs::read_dir(format!("{CURRENT_PATH}/tx-fees-data")).unwrap() {
+        let _version = file
+            .unwrap()
+            .path()
+            .file_name()
+            .unwrap()
+            .to_str()
+            .unwrap()
+            .split("-tx-fees.csv")
+            .collect::<Vec<&str>>()[0]
+            .to_string();
+        if _version.contains("README.md") {
+            continue;
+        }
+        let _version = version_compare::Version::from(&_version).unwrap();
+        let version = version_compare::Version::from(&latest_version).unwrap();
+        if version < _version {
+            latest_version = _version.to_string();
+        }
+    }
+    let version = version_compare::Version::from(&latest_version).unwrap();
+    let csv_path = format!("{CURRENT_PATH}/tx-fees-data/{version}-tx-fees.csv");
+    let mut rdr = csv::Reader::from_path(csv_path).unwrap();
+
+    let all_extrinsics_gas_fees = calculate_all_current_extrinsic_gas_fee();
+
+    let mut last_release_gas_fees = rdr.deserialize().map(|e| {
+        let record: GasFeeDetail = e.unwrap();
+        record
+    });
+
+    for GasFeeDetail {
+        module,
+        extrinsic,
+        gas_fee,
+    } in all_extrinsics_gas_fees.iter()
+    {
+        match last_release_gas_fees.find(|e| e.extrinsic.eq(extrinsic)) {
+            Some(found) => {
+                let fluctuation = Percent::from_float((gas_fee - found.gas_fee).abs() / found.gas_fee);
+                assert!(fluctuation <= GAS_FEE_FLUCTUATION, "The tx fee fluctuation for the extrinsic {extrinsic} is {:?}, bigger than {:?}.", fluctuation, GAS_FEE_FLUCTUATION);
+            }
+            None => panic!("The extrinsic {module}.{extrinsic} is missing from current tx fees list, please add it to latest csv file."),
+        }
+    }
+}
+
+#[test]
+#[ignore]
+fn write_all_current_extrinsic_gas_fee_to_csv() {
+    const VERSION: &str = env!("CARGO_PKG_VERSION");
+    const CURRENT_PATH: &str = env!("CARGO_MANIFEST_DIR");
+    let csv_path = format!("{CURRENT_PATH}/tx-fees-data/{VERSION}-tx-fees.csv");
+
+    let mut wtr = csv::Writer::from_path(csv_path).unwrap();
+    let all_extrinsics_gas_fees = calculate_all_current_extrinsic_gas_fee();
+
+    for extrinsic in all_extrinsics_gas_fees {
+        wtr.serialize(extrinsic).unwrap();
+    }
+    wtr.flush().unwrap();
+}
+
+fn calculate_all_current_extrinsic_gas_fee() -> Vec<GasFeeDetail> {
+    let multiplier = MinimumMultiplier::get();
```

Comment:
The diff is going to be small if we use `MinimumMultiplier` as most of the cost is in base fee

Reply:
It's not just base fee, this line of code will calculate the whole tx fee according to current `MinimumMultiplier`
```rust
let fee = TransactionPayment::compute_fee(call_len, &dispatch_info, 0);
```
If I'm wrong, @ghzlatarev please correct me.

Reply:
Yeah correct but `MinimumMultipler` makes it so almost the entire cost of the tx is in the base fee, if the `Multiplier` increases from congestion the weight of the extrinsic becomes much more heavily weighted in the calculation of the tx fee

Reply:
`Multiplier::one()` seems more reasonable, i agree